### PR TITLE
DM-17489: Write bfKernel and defects in gen2convert along with camera.

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: tests/test_gen2convert.py [hunk 1]
```diff
@@ -116,6 +118,41 @@ def testRawFilters(self):
         rawI = self.butler.get("raw", instrument="HSC", exposure=903986, detector=16)
         self.assertEqual(rawI.getFilter().getName(), "i")
 
+    def testCuratedCalibrations(self):
+        """Test that defects, the camera, and the brighter-fatter kernel were
+        added to the Gen3 registry.
+        """
+        originInfo = DatasetOriginInfoDef(["raw", "calib"], [])
+        rowsWithCalibs = list(
```
Comment (SimonKrughoff) [inline tests/test_gen2convert.py]: Aren't these more like column names?

This is a nit pick, the term `WithCalibs` and `WithoutCalibs` is a bit confusing.  It seems more like `AllDatasets` and `RawOnly`.  Those may not be any better.  Take this with a grain of salt.

File: tests/test_gen2convert.py [hunk 2]
```diff
@@ -116,6 +118,41 @@ def testRawFilters(self):
...
+        )
+        rowsWithoutCalibs = list(
+            self.butler.registry.selectDimensions(
+                originInfo, expression="",
+                neededDatasetTypes=["raw"],
+                futureDatasetTypes=[],
+            )
+        )
+        self.assertGreater(len(rowsWithoutCalibs), 0)
+        self.assertEqual(len(rowsWithCalibs), len(rowsWithoutCalibs))
```
Comment (SimonKrughoff) [inline tests/test_gen2convert.py]: This is just ignorance on my part, but I'm surprised that the two sets should be the same.

Maybe this is a misunderstanding from me.  The name of the method seems like it is trying to select the columns that represent the spanning set of dimensions that describe the set of datasets, but maybe that's not actually what it does.

Comment (TallJimbo) [reply tests/test_gen2convert.py]: It's actually doing something a bit more complex - `selectDimensions` iterates over the data IDs that are jointly valid for all of the given datasetTypes, so in essence we're comparing the total number of raws in the database with the number of raws that have a `camera`, a `bfKernel`, and a set of `defects`.  I'll add some comments to that effect.

File: tests/test_gen2convert.py (hunk 1 above)
Comment (TallJimbo) [reply tests/test_gen2convert.py]: They really are rows, but I agree it's not clear at all what they're rows of (the answer is a query that returns something like a big data ID that has keys for all of the given dataset types). I think it's best to leave the names and add some comments.

Comment (SimonKrughoff) [reply tests/test_gen2convert.py]: OK

File: tests/test_gen2convert.py (hunk 2 above)
Comment (SimonKrughoff) [reply tests/test_gen2convert.py]: OK, great, and thanks.

**Pull request merged**